File: C++/Greedy/체육복.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> value(n, 0);

    for (int i = 0; i < lost.size(); i++)
        value[lost[i] - 1]--;

    for (int i = 0; i < reserve.size(); i++)
        value[reserve[i] - 1]++;
    
    for (int i = 0; i < value.size(); i++) {
        if (value[i] == -1) {
            if (i != 0 && value[i - 1] == 1) {
                value[i - 1]--;
                value[i]++;
            }
            if (i != value.size() - 1 && value[i + 1] == 1) {
                value[i + 1]--;
                value[i]++;
            }
        }
    }

    for (int i = 0; i < value.size(); i++) {
        if (value[i] != -1) answer++;
    }

    return answer;
}

File: Python/Greedy/체육복.py
def solution(n, lost, reserve):
    # students who can borrow
    can_borrow = list(set(reserve) - set(lost))
    # students who need borrow
    need_borrow = list(set(lost) - set(reserve))
    
    answer = n - len(need_borrow)
    
    # start borrowing
    for student in need_borrow:
        if student-1 in can_borrow:
            can_borrow.remove(student-1)
            answer += 1
        elif student+1 in can_borrow:
            can_borrow.remove(student+1)
            answer += 1
    
    return answer
